openedx_proversity_reports/reports/enrollment_report.py

```diff
@@ -0,0 +1,82 @@
+"""
+Enrollment report Report Class.
+"""
+import logging
+from datetime import datetime
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from openedx_proversity_reports.edxapp_wrapper.get_student_account_library import get_user_salesforce_contact_id
+from openedx_proversity_reports.edxapp_wrapper.get_student_library import get_course_enrollment, get_user_profile
+
+
+LOG = logging.getLogger(__name__)
+
+
+class EnrollmentReport(object):
+    """
+    EnrollmentReport Class.
+    """
+
+    def __init__(self, course_id):
+        try:
+            self.course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            LOG.error('Invalid course_id = %s for learning tracker report', course_id)
+            raise InvalidKeyError
+
+    def generate_report(self, **kwargs):
+        """
+        Returns a List with the enrollments for the given dates.
+        """
+        updated_at = kwargs.get('updated_at', '')
+        oldest = kwargs.get('oldest', '')
+        latest = kwargs.get('latest', '')
+        course_enrollment_objects = get_course_enrollment().objects
+        report_data = []
+
+        if updated_at:
+            updated_at = datetime.strptime(updated_at, '%Y-%m-%d')
+            course_enrollments = course_enrollment_objects.filter(
+                created__year=updated_at.year,
+                created__month=updated_at.month,
+                created__day=updated_at.day,
+                course_id=self.course_key,
+            )
+        elif oldest and latest:
+            course_enrollments = course_enrollment_objects.filter(
+                created__range=[oldest, latest],
+                course_id=self.course_key,
+            )
+        elif oldest:
+            course_enrollments = course_enrollment_objects.filter(
+                created__gte=oldest,
+                course_id=self.course_key,
+            )
+        elif latest:
+            course_enrollments = course_enrollment_objects.filter(
+                created__lte=latest,
+                course_id=self.course_key,
+            )
+        else:
+            course_enrollments = course_enrollment_objects.filter(course_id=self.course_key)
```

Comment:
We must return only the course enrollments with users who doesn't have contact id, otherwise, this will always return all the course enrollments, and we don't need it. @andrey-canon